kiota_http/kiota_client_factory.py

```diff
@@ -1,52 +1,63 @@
-import functools
-from sqlite3 import connect
+from __future__ import annotations
+
+from typing import List, Optional
 
 import httpx
 
 from .middleware import (
     AsyncKiotaTransport,
+    BaseMiddleware,
     MiddlewarePipeline,
     ParametersNameDecodingHandler,
     RedirectHandler,
     RetryHandler,
 )
 
+DEFAULT_CONNECTION_TIMEOUT: int = 30
+DEFAULT_REQUEST_TIMEOUT: int = 100
+
 
 class KiotaClientFactory:
-    DEFAULT_CONNECTION_TIMEOUT: int = 30
-    DEFAULT_REQUEST_TIMEOUT: int = 100
 
-    def create_with_default_middleware(self) -> httpx.AsyncClient:
+    @staticmethod
+    def create_with_default_middleware() -> httpx.AsyncClient:
         """Constructs native HTTP AsyncClient(httpx.AsyncClient) instances configured with
         a custom transport loaded with a default pipeline of middleware.
 
         Returns:
             httpx.AsycClient: An instance of the AsyncClient object
         """
-        timeout = httpx.Timeout(
-            self.DEFAULT_REQUEST_TIMEOUT, connect=self.DEFAULT_CONNECTION_TIMEOUT
-        )
+        timeout = httpx.Timeout(DEFAULT_REQUEST_TIMEOUT, connect=DEFAULT_CONNECTION_TIMEOUT)
         kiota_async_client = httpx.AsyncClient(timeout=timeout, http2=True)
         current_transport = kiota_async_client._transport
-        middleware = self._get_default_middleware(current_transport)
+        middleware = KiotaClientFactory._get_default_middleware()
+        middleware_pipeline = KiotaClientFactory._create_middleware_pipeline(
+            middleware, current_transport
+        )
 
         kiota_async_client._transport = AsyncKiotaTransport(
-            transport=current_transport, middleware=middleware
+            transport=current_transport, pipeline=middleware_pipeline
         )
         return kiota_async_client
 
-    def _get_default_middleware(self, transport: httpx.AsyncBaseTransport) -> MiddlewarePipeline:
+    @staticmethod
+    def _get_default_middleware() -> List[BaseMiddleware]:
```

Comment:
I believe this method and the next one should be public in case somebody wants to get the defaults, tweak it (inject an additional one for example), and build a client